src/Tests/CronJobFormTest.php

```diff
@@ -163,24 +165,34 @@ function testManageJob() {
 
     $this->drupalGet('admin/config/system/cron/jobs');
 
-    // Test deleting a job.
-    $this->clickLink(t('Delete'), 1);
-    $this->drupalPostForm(NULL, NULL, t('Delete'));
-    $this->assertText('The cron job edited job name has been deleted.');
-    $this->drupalGet('admin/config/system/cron/jobs');
-    $this->assertNoText($this->job_name);
+    // Save new job.
     $this->clickLink(t('Add job'));
     $job_configuration = array(
       'title' => 'Test Job',
       'id' => strtolower($this->randomMachineName()),
       'scheduler[id]' => 'crontab',
     );
-
-    // Save new job.
     $this->drupalPostForm(NULL, $job_configuration, t('Save'));
-    $this->clickLink(t('Edit'), 1);
-    $this->drupalPostForm(NULL, ['scheduler[configuration][rules][0]' => '0+@ * * * *'], t('Save'));
+    $this->drupalPostForm('admin/config/system/cron/jobs/manage/' . $job_configuration['id'], ['scheduler[configuration][rules][0]' => '0+@ * * * *'], t('Save'));
     $this->assertText('Rule: 0+@ * * * *');
+
+    // Save an invalid cron job.
+    $this->clickLink(t('Edit'), 1);
+    $job = CronJob::load('system_cron');
+    $job->setCallback('dumb_job')->save();
+    $this->drupalGet('admin/config/system/cron/jobs');
+
+    // Assert that the invalid cron job is displayed properly.
+    $this->assertFieldByXPath('//table/tbody/tr[1]/td[6]', 'Missing');
+    $this->assertFieldByXPath('//table/tbody/tr[1]/td[7]/div/div/ul/li/a', 'Delete');
+
+    // Test deleting a job (only possible if invalid cron job).
+    $this->clickLink(t('Delete'), 0);
```

Comment:
We should also check that there is no Delete link on this page before making one invalid.